src/overlays/actors/ovl_En_Bubble/z_en_bubble.c

```diff
@@ -23,45 +26,393 @@ const ActorInit En_Bubble_InitVars = {
+    Collider_DestroyJntSph(globalCtx, &this->colliderSphere);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CC648.s")
+void EnBubble_Wait(EnBubble* this, GlobalContext* globalCtx) {
+    if (EnBubble_DetectPop(this, globalCtx) != 0) {
+        this->explosionCountdown = func_809CBCBC(this);
+        this->actionFunc = EnBubble_Pop;
+    } else {
+        EnBubble_Fly(this, globalCtx);
+        this->actor.shape.unk_08 = ((this->expansionHeight + 1.0f) * 16.0f);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderSphere.base);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliderSphere.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CC6E4.s")
+void EnBubble_Pop(EnBubble* this, GlobalContext* globalCtx) {
+    if (EnBubble_Explosion(this, globalCtx) >= 0) {
+        Audio_PlaySoundAtPosition(globalCtx, &this->actor.posRot.pos, 60, NA_SE_EN_AWA_BREAK);
+        Actor_Kill(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CC774.s")
+// unused
+void EnBubble_Disappear(EnBubble* this, GlobalContext* globalCtx) {
+    s32 temp_v0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/EnBubble_Init.s")
+    temp_v0 = func_809CBFD4(this);
+    if (temp_v0 >= 0) {
+        this->actor.shape.shadowDrawFunc = ActorShadow_DrawFunc_Circle;
+        this->explosionCountdown = temp_v0;
+        this->actionFunc = EnBubble_Regrow;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/EnBubble_Destroy.s")
+// unused
+void EnBubble_Regrow(EnBubble* this, GlobalContext* globalCtx) {
+    if (func_809CC020(this) != 0) {
+        this->actionFunc = EnBubble_Wait;
+    }
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderSphere.base);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliderSphere.base);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CC9B4.s")
+void EnBubble_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnBubble* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CCA60.s")
+    func_8002D7EC(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, 16.0f, 16.0f, 0.0f, 7);
+    this->actionFunc(this, globalCtx);
+    Actor_SetHeight(&this->actor, this->actor.shape.unk_08);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CCAA8.s")
+void EnBubble_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnBubble* this = THIS;
+    u32 pad;
+    GraphicsContext* gfxCtx;
+    Gfx* dispRefs[4];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CCAF0.s")
+    gfxCtx = globalCtx->state.gfxCtx;
+    Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_en_bubble.c", 1175);
```

Comment:
oh youre a boomer decomper now who doesnt know about OPEN_DISPS huh Kappa
this is all a macro now. youll want to use the POLY_XLU_DISP macros to use the gfxCtx temp that it returns

Reply:
Giving a whole new meaning to OK Boomer